### PR TITLE
[HER-163] Add nr of detections on camera tooltip

### PR DESCRIPTION
**Pull request by andreea-dumitrascu**
## Description
We should add the number of camera detections on each camera’s tooltip. 

## Related
[HER-163](https://nerds-interns.atlassian.net/jira/software/projects/HER/boards/1?selectedIssue=HER-163)

### CODE REVIEW

File: src/cameras/map/hooks/use-detections-count.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { useDetections } from 'cameras/map/hooks/use-detections'
+import { useCallback } from 'react'
```
Comment (miutamihai) [inline src/cameras/map/hooks/use-detections-count.js:1]: Let's reorder our imports here.

File: src/cameras/map/hooks/use-init-items.js [hunk 2]
```diff
@@ -2,11 +2,14 @@ import { useEffect } from 'react'
 
 import { useContext } from 'cameras/context'
 import { useItems } from 'cameras/map/hooks/use-items'
+import { useDetectionsCount } from 'cameras/map/hooks/use-detections-count'
 
 export const useInitItems = () => {
   const { setItems } = useContext()
   const items = useItems()
+  const numb = useDetectionsCount()
```
Comment (miutamihai) [inline src/cameras/map/hooks/use-init-items.js]: It is me who's going numb reading this :)))

Comment (miutamihai) [reply src/cameras/map/hooks/use-init-items.js]: Joke aside, let's rename this.

File: src/cameras/map/hooks/use-init-items.js [hunk 3]
```diff
@@ -2,11 +2,14 @@ import { useEffect } from 'react'
...
+
   useEffect(() => {
-    items.length && setItems(items.map(item => ({ ...item, show: false })))
-  }, [setItems, items])
+    items.length && setItems(items.map(camera => ({ ...camera, show: camera.show || false, detections: numb(camera) })))
```
Comment (miutamihai) [inline src/cameras/map/hooks/use-init-items.js]: Also, this `camera.show || false` thingy can be extracted into a separate function (I don't mean separate file, though).

File: src/cameras/map/marker/sort-items.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const sortItems = items => {
```
Comment (miutamihai) [inline src/cameras/map/marker/sort-items.js]: image

This means we can return `items.sort...` directly and turn this into a one liner.

**Review by miutamihai — CHANGES_REQUESTED** [review]

File: src/cameras/map/hooks/use-detections-count.js (hunk 1 above)
Comment (andreea-dumitrascu) [reply src/cameras/map/hooks/use-detections-count.js:1]: Done

File: src/cameras/map/hooks/use-init-items.js (hunk 2 above)
Comment (andreea-dumitrascu) [reply src/cameras/map/hooks/use-init-items.js]: Done

File: src/cameras/map/hooks/use-init-items.js (hunk 3 above)
Comment (andreea-dumitrascu) [reply src/cameras/map/hooks/use-init-items.js]: Done

File: src/cameras/map/marker/sort-items.js (hunk 4 above)
Comment (andreea-dumitrascu) [reply src/cameras/map/marker/sort-items.js]: Done

File: src/cameras/map/hooks/use-init-items.js (hunk 2 above)
Comment (andreea-dumitrascu) [reply src/cameras/map/hooks/use-init-items.js]: :)))

**Review by miutamihai — APPROVED** [review]

**Review by Urzica-Platon-Casian — APPROVED** [review]

**Pull request merged**